selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/theia/TheiaHostedPluginSelectPathForm.java

```diff
@@ -0,0 +1,220 @@
+
+@Singleton
+public class TheiaHostedPluginSelectPathForm {
+  private final SeleniumWebDriverHelper seleniumWebDriverHelper;
+
+  @Inject
+  TheiaHostedPluginSelectPathForm(SeleniumWebDriverHelper seleniumWebDriverHelper) {
+    this.seleniumWebDriverHelper = seleniumWebDriverHelper;
+  }
+
+  public interface Locators {
+    String TITLE_XPATH =
+        "//div[@class='dialogBlock']//div[@class='dialogTitle']//div[text()='Hosted Plugin: Select Path']";
+    String CLOSE_ICON_XPATH =
+        "//div[@class='dialogBlock']//div[@class='dialogTitle']//i[contains(@class, 'closeButton')]";
+    String DROP_DOWN_TITLE_XPATH =
+        "//div[@class='dialogBlock']//select[@class='theia-LocationList']";
+    String DROP_DOWN_SUGGESTIONS_XPATH = DROP_DOWN_TITLE_XPATH + "//option";
+    String DROP_DOWN_SUGGESTION_XPATH_TEMPLATE =
+        "(//div[@class='dialogBlock']//select[@class='theia-LocationList']//option)[%s]";
+
+    String ROOT_PROJECTS_FOLDER_XPATH =
+        "//div[@class='dialogBlock']//div[@class='theia-TreeNodeContent']//div[@id='/projects']";
+    String PROJECT_TREE_ITEM_ID_TEMPLATE = "/projects/%s";
+    String SELECTED_TEM_XPATH_TEMPLATE =
+        "//div[@data-node-id='/projects/%s']/parent::div/parent::div[contains(@class, 'theia-mod-selected') and contains(@class, 'theia-mod-focus')]";
+    String CANCEL_BUTTON_XPATH = "//div[@class='dialogBlock']//button[text()='Cancel']";
+    String OPEN_BUTTON_XPATH = "//div[@class='dialogBlock']//button[text()='Open']";
+  }
+
+  private String getSelectedItemXpath(String itemPath) {
+    return format(SELECTED_TEM_XPATH_TEMPLATE, itemPath);
+  }
+
+  private String getProjectsTreeItemId(String itemPath) {
+    return format(PROJECT_TREE_ITEM_ID_TEMPLATE, itemPath);
+  }
+
+  private String getDropDownSuggestionXpath(int suggestionIndex) {
+    final int adoptedSuggestionIndex = suggestionIndex + 1;
+    return format(DROP_DOWN_SUGGESTION_XPATH_TEMPLATE, adoptedSuggestionIndex);
+  }
+
+  public void waitTitle() {
+    seleniumWebDriverHelper.waitVisibility(By.xpath(TITLE_XPATH));
+  }
+
+  public void waitTitleDissappearance() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath(TITLE_XPATH));
+  }
+
+  public void waitCloseIcon() {
+    seleniumWebDriverHelper.waitVisibility(By.xpath(CLOSE_ICON_XPATH));
+  }
+
+  public void clickOnCloseIcon() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(CLOSE_ICON_XPATH));
+  }
+
+  public String getDropDownTitleText() {
+    return seleniumWebDriverHelper.waitVisibilityAndGetValue(By.xpath(DROP_DOWN_TITLE_XPATH));
+  }
+
+  public void waitDropDownTitleText(String expectedText) {
+    seleniumWebDriverHelper.waitValueEqualsTo(By.xpath(DROP_DOWN_TITLE_XPATH), expectedText);
+  }
+
+  private int getDropDownSuggestionsCount() {
+    return seleniumWebDriverHelper
+        .waitVisibilityOfAllElements(By.xpath(DROP_DOWN_SUGGESTIONS_XPATH))
+        .size();
+  }
+
+  public List<String> getDropDownSuggestions() {
```

Comment:
IMHO, if there is only one way to obtain suggestions - from dropdown menu, you can get rid of `DropDown` word as following:
- `getDropDownSuggestions` > `getSuggestions()`
- `waitDropDownTitleText` > `waitSuggestionsTitleText`
- `DROP_DOWN_SUGGESTION_XPATH_TEMPLATE` > `SUGGESTION_XPATH_TEMPLATE`